src/app/core/service/dialog/dialog-ref.service.ts

```diff
@@ -29,33 +31,46 @@ export class DialogRefService {
   constructor(private readonly shortcutService: ShortcutService) {}
 
   public register(): void {
-    this.escapeSubscription = this.shortcutService
-      .add(
-        'escape',
-        false,
-        VisibleFlag.Game | VisibleFlag.Dialog,
-        VisibleFlag.Overlay | VisibleFlag.Dialog,
-        VisibleFlag.Browser
-      )
-      .subscribe(() => this.close())
-
-    this.spaceSubscription = this.shortcutService
-      .add(
-        'space',
-        false,
-        VisibleFlag.Game | VisibleFlag.Dialog,
-        VisibleFlag.Overlay | VisibleFlag.Dialog
-      )
-      .subscribe(() => this.closeAll())
-  }
+    if (!this.escapeSubscription) {
+      const clearShortcut = () => {
+        this.escapeSubscription?.unsubscribe()
+        this.escapeSubscription = null
+      }
```

Comment:
this seems weird. you are only clearing the subscription if the subscription is falsey?

Reply:
The `clearShortcut` method that's defined here, is passed on (to another method) that'll call it some time in the future. At which point this (or any other class that has access) could have already cleared the shortcut manually, that's why the check with `?` is inside the method.  
This entire if-block is here to only create/register the shotcut once, regardless of how many times the method it resides in is called.